pkg/controller/controller.go

```diff
@@ -624,6 +670,25 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	return err
 }
 
+func (p *csiProvisioner) SupportsBlock() bool {
+	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
+	defer cancel()
+
+	client := csi.NewControllerClient(p.grpcClient)
+	req := csi.ValidateVolumeCapabilitiesRequest{
+		VolumeCapabilities: []*csi.VolumeCapability{
+			volumeCapabilityBlockReadWriteOnce,
+		},
+	}
```

Comment:
Thank you for pointing this out.

The intention of ```SupportsBlock``` is to check supportability of block volume feature for the driver before ```Provision``` is called so that provisioner can skip ```Provision```, if the driver doesn't support block volume feature. (Drivers like Ceph, NFS, and hostpath that doesn't provide block volume would need this.) Therefore, if ```ValidateVolumeCapabilitiesRequest``` requires a volume ID, it won't be used here, because the existence of volume ID requires existence of volume.

So, the alternatives would be below three:

1. Just return ```true``` for ```SupportsBlock``` in all the driver and make ```Provision``` fail for block unsupported drivers.
2. Change CSI spec to allow ```ValidateVolumeCapabilitiesRequest``` not specifying volumeID and call ```ValidateVolumeCapabilitiesRequest``` here
3. Add another method in CSI spec to be used for this use case, and call the new method here.

Any thoughts?

Reply:
I like alternative 1. The CSI spec doesn't let you query abstract capabilities, and 2 or 3 require spec changes which are unlikely to happen before the promotion to v1.

Reply:
@bswartz

Thank you for your comment. I will rebase the implementation for alternative 1 and push here, if @vladimirvivien also agrees.

Reply:
When you rebase, you'll have a conflict with the mount options work I did. I can help fix those conflicts if you like. I'm currently testing this PR in my lab to see if I can get it to work.